system/BaseModel.php

```diff
@@ -286,6 +286,11 @@ abstract class BaseModel
      */
     protected $afterDelete = [];
 
+    /**
+     * Whether to prohibit inserting empty data.
+     */
+    protected bool $prohibitInsertEmpty = true;
```

Comment:
Consider naming after the existing properties. 
```suggestion
    protected bool $allowEmptyInserts = false;
```

Reply:
Done.